lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -134,7 +141,14 @@ static public function convertPositionalToNamedPlaceholders($statement)
      */
     public function bindValue($param, $value, $type = null)
     {
-        return $this->bindParam($param, $value, $type, null);
+        if ($value instanceof OCI8Descriptor) {
```

Comment:
Binding should work the same in PHP 5 & 7 as long as the bound zval still exists when `oci_execute()` is later called.  This has always been the case.  But the PHP 5 VM held an extra refcount on zvals which PHP 7 doesn't.  This means that cleanup can occur in PHP 7 before `oci_execute()` is called.  The OCI8 driver can't fake the refcount since it doesn't know what the app wants to do with the var.  Incrementing the refcount can fix some IN bind cases but will break OUT binds.  The bottom line is that the app has to manage the refcount and end-of-scope cleanup.  If you are seeing any bind issues where bind & execute are done in the same scope, log a bug.